cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java

```diff
@@ -153,49 +138,42 @@
   // This list is to record ApplicationManager create inside @Test method
   private final List<ApplicationManager> applicationManagers = Lists.newArrayList();
 
+  private static TestManager testManager;
+
+  protected static TestManager getTestManager() {
+    Preconditions.checkState(testManager != null, "Test framework is not yet running");
+    return testManager;
+  }
+
   /**
    * Deploys an {@link Application}. The {@link co.cask.cdap.api.flow.Flow Flows} and
    * {@link co.cask.cdap.api.procedure.Procedure Procedures} defined in the application
    * must be in the same or children package as the application.
    *
+   * @deprecated Use {@link TestManager#deployApplication(Class, java.io.File...)} from {@link #getTestManager()}.
+   *
    * @param applicationClz The application class
    * @return An {@link co.cask.cdap.test.ApplicationManager} to manage the deployed application.
    */
+  @Deprecated
   protected ApplicationManager deployApplication(Class<? extends Application> applicationClz,
-                                                 File...bundleEmbeddedJars) {
-    
-    Preconditions.checkNotNull(applicationClz, "Application class cannot be null.");
-
-    try {
-      Application app = applicationClz.newInstance();
-      DefaultAppConfigurer configurer = new DefaultAppConfigurer(app);
-      app.configure(configurer, new ApplicationContext());
-      ApplicationSpecification appSpec = configurer.createSpecification();
-
-      Location deployedJar = appFabricClient.deployApplication(appSpec.getName(), applicationClz, bundleEmbeddedJars);
-      ApplicationManager manager = injector.getInstance(ApplicationManagerFactory.class)
-                                           .create(DefaultId.NAMESPACE.getId(), appSpec.getName(),
-                                                   deployedJar, appSpec);
-      applicationManagers.add(manager);
-      return manager;
-
-    } catch (Exception e) {
-      throw Throwables.propagate(e);
-    }
+                                                 File... bundleEmbeddedJars) {
+    Preconditions.checkState(testManager != null, "Test framework is not yet running");
+    ApplicationManager applicationManager = testManager.deployApplication(applicationClz, bundleEmbeddedJars);
```

Comment:
Should always call `getTestManager()`, so that in future if we want to do anything special in the method (e.g. some lazy creation stuff), it's centralized.